jcapi-systems.go

```diff
@@ -221,10 +232,15 @@ func (jc JCAPI) GetSystemByHostName(hostname string, withTags bool) ([]JCSystem,
 	var returnVal []JCSystem
 
 	jcSystemRec, err := jc.Post("/search/systems", jc.hostnameFilter(hostname))
+
```

Comment:
The more common formatting in Go is to check for errors on the immediate next line. I see you've done it this way a lot, but just letting you know that this is a non-standard formatting in general.